change(wpa_supplicant): Replace use of mbedtls_ctr_drbg with esp_mbedtls_random()

rojer opened the pull request:
## Description

This way wpa_supplicant does not require ctr_drbg that saves some space

## Related

Requires https://github.com/espressif/esp-idf/pull/15678

## Testing

Compiles and works, been using it for a long time

## Checklist

Before submitting a Pull Request, please ensure the following:

- [x] 🚨 This PR does not introduce breaking changes.
- [x] All CI checks (GH Actions) pass.
- [x] Documentation is updated as needed.
- [x] Tests are updated or added as necessary.
- [x] Code is well-commented, especially in complex areas.
- [x] Git history is clean — commits are squashed to the minimum necessary.

kapilkedawat commented:
mbedtls funtions should be defined in mbedtls component, we can retain crypto_rng_wrapper() and just have the changes related to ctr_drbg

rojer commented:
thanks for the review @kapilkedawat , comments addressed:

> mbedtls funtions should be defined in mbedtls component

ok, updated https://github.com/espressif/esp-idf/pull/15678 - moved to `mbedtls/esp_random.h` and renamed to `mbedtls_esp_random`.

> we can retain crypto_rng_wrapper() and just have the changes related to ctr_drbg

well, `crypto_rng_wrapper` is just `mbedtls_esp_random` repeated multiple times in different files - i thought i might as well clean it up. but if you insist, i can keep it.

kapilkedawat commented on components/wpa_supplicant/esp_supplicant/src/crypto/crypto_mbedtls-bignum.c:
```diff
@@ -16,11 +17,6 @@
-{
-    return random_get_bytes(buf, len);
-}
-
```
let's keep this, shouldn't be much overhead.

rojer replied:
it's not much, but it is literally the same, so why keep it, now that we have `mbedtls_esp_random`?

rojer replied:
@kapilkedawat so what do you think?

kapilkedawat commented on components/wpa_supplicant/esp_supplicant/src/crypto/crypto_mbedtls-ec.c:
```diff
@@ -36,10 +35,9 @@
-}
+struct crypto_ec {
+	mbedtls_ecp_group group;
+};
```
no need of this change, current implementation is cleaner.

rojer replied:
current `crypto_rng_wrapper` is literally the same as `mbedtls_esp_random`, so i'm just removing the duplication. in what way is it cleaner?

kapilkedawat commented:
Sorry, missed this PR, generally looks good, gave some comments.. We can remove second commit.

kapilkedawat commented:
https://github.com/espressif/esp-idf/pull/15678 won't be needed if we implement suggested changes here, right?

rojer commented:
> #15678 won't be needed if we implement suggested changes here, right?

oh, i get it, you didn't mean get rid of a commit on this MR but or the other MR. sure, that's fine, doesn't matter much. i'll keep the commit distinct but will focus on this MR.

rojer commented:
@kapilkedawat ok, i rebased the branch and fixed an inadvertent earlier rebase issue related to `struct ec`. pre-commit checks are now passing.

i still don't see a point of keeping `crypto_rng_wrapper` since it just duplicates `mbedtls_esp_random` but if you insist i can bring it back. let me know.

rojer commented:
ping @kapilkedawat  ?

kapilkedawat commented:
@rojer this is raised internally and review is ongoing.

rojer commented:
@kapilkedawat any update?
